### PR TITLE
[ANCHOR-483] Add metrics for SEP-6

### PR DESCRIPTION
**Pull request by philipliu**
### Description

This adds metrics to SEP-6.

### Context

Observability

### Testing

- `./gradlew test`

### Documentation

N/A

### Known limitations

N/A

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/service/AnchorMetrics.java [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package org.stellar.anchor.platform.service;
 
 public enum AnchorMetrics {
+  SEP6_TRANSACTION_OBSERVED("sep6.transaction.observed"),
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/service/AnchorMetrics.java:4]: WDYT to merge `AnchorMetrics.java` with `MetricsConstants.java`?

File: platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java [hunk 2]
```diff
@@ -257,5 +257,13 @@ void handleSep6Transaction(ObservedPayment payment, JdbcSep6Transaction txn)
         payment.getTransactionHash(),
         payment.getAmount(),
         rpcConfig.getCustomMessages().getIncomingPaymentReceived());
+
+    Metrics.counter(
+            AnchorMetrics.SEP6_TRANSACTION_OBSERVED.toString(),
+            "status",
+            SepTransactionStatus.PENDING_ANCHOR.toString())
+        .increment();
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java:265]: We have two metrics patterns. 

WDYT to use something like:
```suggestion
Counter sep6WithdrawalCounter =
      counter(
          MetricConstants.SEP6_TRANSACTION_OBSERVED
          MetricConstants.STATUS,
          MetricConstants.PENDING_ANCHOR);
sep6WithdrawalCounter.increment();
```

File: platform/src/main/java/org/stellar/anchor/platform/service/AnchorMetrics.java (hunk 1 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/service/AnchorMetrics.java:4]: I think the metrics need to be refactored quite a bit. If it's all the same to you, I would like to do this in a separate PR.

File: platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java (hunk 2 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java:265]: I think we should probably differentiate tags and metrics either by type or by making them members of different classes.

File: platform/src/main/java/org/stellar/anchor/platform/service/AnchorMetrics.java (hunk 1 above)
Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/service/AnchorMetrics.java:4]: It's ok to be in a separate PR.

File: platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java (hunk 2 above)
Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java:265]: Do it in a separate PR?

Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java:265]: Yep

**Pull request merged**